src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -755,7 +851,16 @@ int
 xrtAIEStartProfiling(xrtDeviceHandle handle, int option, const char *port1Name, const char *port2Name, uint32_t value)
 {
   try {
-    return start_profiling(handle, option, port1Name, port2Name, value);
+    auto event = create_profiling_event(handle);
+    if (option < 0 || option > 3)
```

Comment:
magic numbers, can you replace this with proper enums? You can fix it in next PR.